er/src/App.jsx

```diff
@@ -283,62 +278,55 @@ const App = (props) => {
             'icon-anchor': 'bottom',
             'icon-allow-overlap': true
           }
-        })
+        });
 
         // Subject Nametag Icon
-        const iconSizeSubjectNametagIcon = !config.map || config.map.simplify_map_data ? ['step', ['zoom'], 1.1, 10, 1.1] : 1.1
-        const iconSizeSubjectNametagText = !config.map || config.map.simplify_map_data ? ['step', ['zoom'], 10, 10, 15] : 15
+        const iconSizeSubjectNametagText = ['step', ['zoom'], 12, 10, 16];
         window.GlobalMap.addLayer({
           id: 'box' + json.id,
           type: 'symbol',
           source: 'point' + json.id,
           layout: {
-            'icon-image': 'subject-popup-box',
-            'icon-size': iconSizeSubjectNametagIcon,
-            'icon-anchor': 'top',
-            'icon-allow-overlap': true,
-            'icon-text-fit': 'both',
-            'icon-text-fit-padding': [3, 3, 3, 3],
             'text-anchor': 'top',
             'text-offset': [0, 0.5],
             'text-allow-overlap': true,
             'text-field': json.name,
-            'text-size': iconSizeSubjectNametagText
+            'text-size': iconSizeSubjectNametagText,
+            visibility: (!config.map || config.map.show_subject_names) ? 'visible' : 'none'
```

Comment:
This `config.map.show_subject_names`-based derivation of label visibility fulfills the "Display subject names on map only when the subject has been selected." requirement by making it configurable.